roles/automation_services_catalog_client/tasks/main.yml

```diff
@@ -1,10 +1,17 @@
 ---
+- name: Set the service_catalog_controller url value from inventory
+  set_fact:
+    _service_catalog_controller_url: "{{ service_catalog_controller_url | default(hostvars[groups['tower'][0]]['ansible_host']) | default(hostvars[groups['tower'][0]]['inventory_hostname']) }}"
+
 - name: Run preflight checks
   include_tasks: preflight.yml
 
+- name: Generate token for catalog worker auth
```

Comment:
@Spredzy If we want to use the same collection for pre 4.0 customers would it make sense to make this conditional. That way a customer could supply the token and we would just use it. Or would this logic work even for pre 4.0 customers.

Reply:
It has been tested on both 3.8 and 4.0 - and it works for both.